### PR TITLE
Update deployment docs

### PR DESCRIPTION
**Pull request by richard-to**
Reorganizes the docs to include Cloud Run, App Engine, and Docker configurations.

### CODE REVIEW

File: docs/guides/deployment.md [hunk 1]
```diff
@@ -31,31 +62,127 @@ The `--bind: 8080` will run Mesop on port 8080.
...
+# Add app code here
+COPY . /srv/mesop-app
+WORKDIR /srv/mesop-app
+
+# Run Mesop through gunicorn. Should be available at localhost:8080
+CMD ["gunicorn", "--bind", "0.0.0.0:8080", "main:me"]
+```
+
+### docker-compose.yaml
+
```
Comment (wwwillchen) [inline docs/guides/deployment.md:169]: I'm not super familiar with the Docker ecosystem, but I feel like it's quite common to launch docker containers directly from the CLI, perhaps include the shell command to directly run the Mesop container without docker compose? e.g.

```
docker build -t mesop-app . && docker run -p 8080:8080 mesop-app
```

**Review by wwwillchen — APPROVED** [review]
Very nice! This will be super helpful as I've seen tons of questions about this on social media. 

One thing that might be nice to mention is deploying onto Hugging face spaces since it's very popular with the ML community. If you have time, it would be great to try the steps in https://huggingface.co/docs/hub/spaces-sdks-docker - in theory, it should be straightforward since you have the docker steps here

File: docs/guides/deployment.md [hunk 2]
```diff
@@ -1,26 +1,57 @@
...
 
-To deploy your Mesop application, we recommend using [Google Cloud Run](https://cloud.google.com/run) because it's easy to get started and there's a [free tier](https://cloud.google.com/run#pricing). However, it's possible to deploy your Mesop to other Cloud platforms.
+This section describes how to run your Mesop application using the following
+platforms:
+
+- [Google Cloud Run](https://cloud.google.com/run)
+- [Google App Engine](https://cloud.google.com/appengine)
+- [Docker](https://www.docker.com/)
+
+It's possible to deploy your Mesop application to other Cloud Platforms as well.
```
[the remark this answers is not in the document]
Comment (richard-to) [reply docs/guides/deployment.md]: Good point. Updated.

**Comment by richard-to** [comment]
I've made a note for deploying to Hugging Face.

**Pull request merged**